types/kagekiri.d.ts

```diff
@@ -1,3 +1,4 @@
 // Type definitions for kagekiri
 export function querySelector(selector: string, context?: Node): HTMLElement | null;
-export function querySelectorAll(selector: string, context?: Node): NodeList;
\ No newline at end of file
+export function querySelectorAll(selector: string, context?: Node): Node[];
+export function getElementsByClassName(names: string, context?: Node): Node[];
```

Comment:
On second thought, it seems that all of these should be `Element` rather than `HTMLElement` or `Node`. Looking at [the MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement), it appears that kagekiri can return Elements (e.g. HTMLElement or SVGElement – you can query SVGs just like HTML), but not generic Nodes (e.g. comments, text).

Reply:
Yeah the documentation is confusing. Apparently `querySelectorAll` [returns](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll#Syntax):

> **A non-live NodeList containing one Element object for each element that matches** at least one of the specified selectors or an empty NodeList in case of no matches.

A NodeList containing Elements? Also, I'm not aware of any way to query non-element nodes (text or comment) using `querySelectorAll`.

What do you want to do? My vote would be for `Node[]` -- IMO it aligns nicely with the spec and what we actually return, but up to you.

Reply:
Despite the name, it seems like `NodeList` actually contains `Element`s, not `Node`s.

I tested, and kagekiri can definitely return `SVGElement` objects (e.g. a `<g>` tag). However, I don't see how it can return generic [Node](https://developer.mozilla.org/en-US/docs/Web/API/Node)s, which include:

> Attr, CharacterData (which Text, Comment, and CDATASection are all based on), ProcessingInstruction, DocumentType, Notation, Entity, and EntityReference

So my vote would be for `Element[]`, since I can't think of any `Node` but non-`Element` that kagekiri can return.